src/marten/db/query/sql/query.cr

```diff
@@ -313,6 +320,36 @@ module Marten
             {sql, parameters}
           end
 
+          private def build_count_column_specific_subquery_query(column_name : String)
+            build_sql do |s|
+              # Because a subquery with the name "subquery" is used here, we cut off the table name at the front
+              s << "SELECT COUNT(#{column_name.split(".")[-1]})"
```

Comment:
I am realizing that relying on additional methods isn't ideal here because the beginning of the count SQL query is defined in the `#build_count_column_specific_subquery_query` / `#build_count_default_subquery_query` methods while the end is still defined in the `#build_count_query` method. This doesn't make it easy to follow the logic of the generation of this SQL query.

Maybe we could put everything in the `#build_count_query` method? I think this would be simpler to follow. 

For example, something along those lines would work IMHO:

```crystal
private def build_count_query(column_name : String | Nil)
  where, parameters = where_clause_and_parameters
  limit = connection.limit_value(@limit)

  sql = build_sql do |s|
    s << "SELECT COUNT(#{column_name ? column_name.split(".")[-1] : nil})"
    s << "FROM ("
    s << "SELECT"

    if distinct
      s << connection.distinct_clause_for(distinct_columns)
    elsif column_name.nil?
      s << "#{Model.db_table}.#{Model.pk_field.db_column!}"
    end

    if !column_name.nil?
      s << column_name
    end

    s << "FROM #{table_name}"
    s << build_joins
    s << where
    s << "LIMIT #{limit}" unless limit.nil?
    s << "OFFSET #{@offset}" unless @offset.nil?
    s << ") subquery"
  end

  {sql, parameters}
end
```

What do you think?

Reply:
I think the number of ifs is reasonable here, but I am also fine with two completely separate methods that take care of generating two distinct (and complete) SQL queries. What I want to avoid ideally (and in order to improve readability) is having a SQL query whose beginning is generated by a method and whose end is generated by another method.